packages/CodeQuality/tests/Rector/Foreach_/ForeachItemsAssignToEmptyArrayToAssignRector/Fixture/skip_false_positives.php.inc

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Rector\CodeQuality\Tests\Rector\Foreach_\ForeachItemsAssignToEmptyArrayToAssignRector\Fixture;
+
+class SkipFalsePositives
+{
+    public function indexBy(array $array, callable $key): array
+    {
+        $indexed = [];
+        foreach ($array as $item) {
+            $indexed[$key($item)] = $item;
```

Comment:
There have to be assign with brackets: `$values[]`

Currently any key is accepted (`$values[$anything]`), see this line in the rule: https://github.com/rectorphp/rector/blob/master/packages/CodeQuality/src/Rector/Foreach_/ForeachItemsAssignToEmptyArrayToAssignRector.php#L107

It is missing check for `$arrayDimFetch->dim === null`, https://github.com/nikic/PHP-Parser/blob/88f3a669c12e576887dfded483e971d6c333adbe/lib/PhpParser/Node/Expr/ArrayDimFetch.php#L12

If the `$arrayDimFetch->dim` value is not null, stop and `return null;`

This should fix this case

Reply:
After this is handled, we'll look on following cases